addons/project/controllers/public_project.py

```diff
@@ -0,0 +1,369 @@
+
+        :param task_id: (int) The ID of the task to track
+        :param fields_to_track: ([str]) The list of fields to track on the task
+        :return: (Object, dict, dict) The task object, tracked fields and their initial values
+        """
+        task = request.env['project.task']
+        task = task.sudo() if not self._is_logged_in() else task
+        task = task.browse(task_id) if task_id else task
+        tracked_fields = task.fields_get(fields_to_track) if task else None
+        initial_values = dict([(task.id, {field: task[field] for field in fields_to_track})])
+        return task, tracked_fields, initial_values
+
+    def _chatter_track(self, task, tracked_fields, initial_values):
+        """Track fields of a task for the chatter, as Public User if user is not logged in.
+
+        :param task: (Object) The task object
+        :param tracked_fields: (dict) The tracked fields
+        :param initial_values: (dict) The tracked fields' initial values
+        :return: True
+        """
+        kwargs = {'author_id': request.env.uid} if not self._is_logged_in() else {}
+        task.message_track(tracked_fields, initial_values, **kwargs)
+        return True
+
+    def _check_and_perform(self, method_to_call, submethod=None, access_info=None, **kw):
+        """Check permissions before calling a given method.
+
+        :param method_to_call: (str) The name of the method to call if access is granted.
+        :param submethod: (str) The name of the method that the method to call will call
+                                if access is granted (if method_to_call == call or call_kw).
+                                Defaults to None.
+        :param kw: (dict) All args and kwargs passed to the method to call.
+        :return: (any) The result of the method to call or None if no access was granted.
+        """
+        read_words = ['read', 'load', 'default', 'get', 'search',
+                     'message', 'mail',  # we want to allow readonly users to use the chatter
+                     'onchange', 'export_data']
+        edit_words = ['write', 'resequence', 'set']
+
+        if any(word in (submethod or method_to_call) for word in read_words):
+            return self._check_and_perform_read(method_to_call,
+                                                submethod=submethod,
+                                                access_info=access_info,
+                                                **kw)
+        if any(word in (submethod or method_to_call) for word in edit_words):
+            return self._check_and_perform_write(method_to_call,
+                                                 submethod=submethod,
+                                                 access_info=access_info,
+                                                 **kw)
+        if 'create' in (submethod or method_to_call):
+            return self._check_and_perform_create(method_to_call,
```

Comment:
(Above comment was by ODO)
Not authorizing operations on the 'res.user' model implies removing the 'assign to' field in task's quick create form. 'assigned to' having a default value, this has the effect of assigning the task to an arbitrary user. What behavior do we wish here?

Reply:
Done. Regarding the domains, I did roughly this:
```python
for d in domain:
    if not isinstance(d, str):
        is_field_ok = self._is_whitelisted(d[0])
        has_subexpr = isinstance(d[0], str) and '.' in d[0]
        if not is_field_ok or has_subexpr:
            raise AccessError("nope")
```
`has_subexpr` is not necessary as no field in the whitelist has a '.' sub-expression, but I included it anyway in case someone ever adds one of those to the whitelist. In any case the fact that I'm checking all d[0]'s against a whitelist of fields makes that I don't need to whitelist domains in particular.